processor/k8sattributesprocessor/e2e_test.go

```diff
@@ -784,8 +925,12 @@ func scanLogsForAttributes(t *testing.T, ls *consumertest.LogsSink, expectedServ
 
 func resourceHasAttributes(resource pcommon.Resource, kvs map[string]*expectedValue) error {
 	foundAttrs := make(map[string]bool)
-	for k := range kvs {
-		foundAttrs[k] = false
+	shouldNotFoundAttrs := make(map[string]bool)
+	for k, v := range kvs {
+		if v.mode != shouldnotexist {
+			foundAttrs[k] = false
```

Comment:
we'd better add `continue` here, since the key in "exist" mode need not to be added to `shouldNotFoundAttrs`